foundationdb/src/api.rs

```diff
@@ -295,34 +471,8 @@ pub struct NetworkStop {
 impl NetworkStop {
     /// Signals the event loop invoked by `Network::run` to terminate.
     pub fn stop(self) -> FdbResult<()> {
-        error::eval(unsafe { fdb_sys::fdb_stop_network() })
-    }
-}
-
-/// Stop the associated `NetworkRunner` and thread if dropped
-///
-/// If trying to stop the FoundationDB run loop results in an error.
-/// The error is printed in `stderr` and the process aborts.
-///
-/// # Panics
-///
-/// Panics if the network thread cannot be joined.
-pub struct NetworkAutoStop {
-    network: Option<NetworkStop>,
-    handle: Option<std::thread::JoinHandle<()>>,
-}
-impl Drop for NetworkAutoStop {
-    fn drop(&mut self) {
-        if let Err(err) = self.network.take().unwrap().stop() {
-            eprintln!("failed to stop network: {err}");
-            // Not aborting can probably cause undefined behavior
-            std::process::abort();
-        }
-        self.handle
-            .take()
-            .unwrap()
-            .join()
-            .expect("failed to join fdb thread");
+        // NetworkStop can only be obtain in unmanaged mode, so stop is guaranteed to return Some<FdbResult>
+        GLOBAL_STATE.lock().unwrap().stop().unwrap()
```

Comment:
the second unwrap should be an expect that explains what happens. The first unwrap is just a mutex poisoning so not as essential to cleanup, but nice.